### ISSUE TITLE
Drag-and-drop for TreeView

### ISSUE THREAD

**Issue by jvanallen93** [vmware/clarity#406]
**Select one ...**  (check one with "x")
```
[ ] bug
[x] feature request
[ ] enhancement
```

### Expected behavior

As a user, I would like to be able to drag and drop elements in the tree view so that I can define a custom tree view.

This is the plugin being reference:
http://dbushell.github.io/Nestable/
* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

**Comment by jaffoneh** [comment]
Assigning to @adityarb88 to take a look and evaluate. CC @youdz

**Comment by youdz** [comment]
I think that it needs to go through UX before we can evaluate. Our current Tree uses no "container" of any kind on the nodes, which makes it hard to understand that they're draggable or even drop targets.

Anything that you can drag should have a very specific visual hint that is consistent across our components, and drop targets should be clearly delimited with some kind of border.

All in all, we might end up looking at a separate component, since they would share no logic, just markup.

**Comment by joshvanal** [comment]
@youdz Completely understood. I was just referencing an implementation when I post that link so it might get some brain gears going on how it might be developed in Clarity.

**Comment by mathisscott** [comment]
@kannan625 
We use Github labels to inform people where items are in the process. This is currently not "In Queue" so it isn't in line to be worked on in the near future. 

You can go [here to see a list of labels that we use and descriptions of what some of them mean](https://github.com/vmware/clarity/labels).

**Comment by kannan625** [comment]
Thanks for the update @mathisscott, appreciated, new to github so not aware of the labels. i will watch for that. Thanks.

**Comment by gnomeontherun** [comment]
Closing this as a duplicate of https://github.com/vmware/clarity/issues/1821
